### PR TITLE
[TOSA] Add aten.clamp op tosa support

### PR DESCRIPTION
**Pull request by AmosLewis**
This issue is found when converting deberta model https://github.com/nod-ai/SHARK/issues/493

### CODE REVIEW

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -3027,6 +3027,37 @@ LogicalResult ConvertAtenOp<AtenWhereSelfOp>::matchAndRewrite(
...
+template <>
+LogicalResult ConvertAtenOp<AtenClampOp>::matchAndRewrite(
+    AtenClampOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+
+  // Not a tensor type.
+  auto selfType = adaptor.self().getType().dyn_cast<TensorType>();
+  if (!selfType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
```
Comment (tatwaichong) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:3039]: nit: A very tiny question. Forget to start the first sentence with a lower-case letter?

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 2]
```diff
@@ -3027,6 +3027,37 @@ LogicalResult ConvertAtenOp<AtenWhereSelfOp>::matchAndRewrite(
...
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: value `int_min` should be a torch constant int");
+
+  if (!matchPattern(op.max(), m_TorchConstantInt(&int_max)))
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: value `int_max` should be a torch constant int");
+
+  IntegerAttr min_int = rewriter.getI64IntegerAttr(int_min);
+  IntegerAttr max_int = rewriter.getI64IntegerAttr(int_max);
+  FloatAttr min_fp = rewriter.getF32FloatAttr(float(int_min));
```
Comment (tatwaichong) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:3052]: I wonder is an explicit integral->floating conversion needed here?

**Review by tatwaichong — COMMENTED** [review]
I come across this PR. I just have a couple side comment on it, and hopefully these are relevant. Overall this is a nice patch.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 1 above)
Comment (eric-k256) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:3039]: good catch, yes, to be consistent, "Only" -> "only"

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 2 above)
Comment (eric-k256) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:3052]: There is the float(int_min) already there.

**Review by eric-k256 — CHANGES_REQUESTED** [review]
Just the one minor message correction needed.

**Review by eric-k256 — APPROVED** [review]
looks good

**Pull request merged**